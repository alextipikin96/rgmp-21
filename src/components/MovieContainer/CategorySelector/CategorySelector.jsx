import React, { useState } from "react";
import "./CategorySelector.scss";

export default ({ selectedCategories, chooseCategoryHandler }) => {
  const categories = [
    "Documentary",
    "Comedy",
    "Horror",
    "Crime",
    "Adventure",
    "Science Fiction",
    "Action",
    "Family",
    "Animation",
    "Fantasy",
    "Drama",
    "Romance",
  ];

  const [isVisible, setVisibility] = useState(false);

  const showAllCategories = () => {
    setVisibility((prevState) => !prevState);
  };

  return (
    <>
      <label form="genre">genre</label>
      <div className="inputWrapper" onClick={showAllCategories}>
        <div className="selectorIcon" />
        <input
          id="genre"
          value={selectedCategories.join(", ")}
          placeholder="Select genre"
          disabled
        />
      </div>
      <ul className="categoryList" hidden={!isVisible}>
        {categories.map((category) => (
          <li className="categoryItem" key={category}>
            <div className="categoryItem__wrapper">
              <input
                id="genres"
                name="genres"
                type="checkbox"
                onChange={() =>
                  chooseCategoryHandler(
                    "genres",
                    selectedCategories.includes(category)
                      ? selectedCategories.slice(0, selectedCategories.length - 1)
                      : selectedCategories.concat(category)
                  )
                }
              />
              <span className="">{category}</span>
            </div>
          </li>
        ))}
      </ul>
    </>
  );
};
