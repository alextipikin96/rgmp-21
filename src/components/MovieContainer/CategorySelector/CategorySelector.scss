@import "../../../assets/scss/colors";

.inputWrapper {
  position: relative;
}

.categoryList {
  padding: 0;
  margin: 0;
  max-height: 190px;
  overflow-x: hidden;
  box-shadow: 0 0 40px 40px rgba(0, 0, 0, 0.2);
}

.categoryItem {
  display: flex;
  padding: 0px 15px;

  &__wrapper {
    position: relative;
    display: flex;
    margin: 10px 0;
    cursor: pointer;
    font-size: 22px;
    user-select: none;

    & span {
      color: $white;
      font-size: 18px;
      margin-left: 10px;
      text-transform: capitalize;
    }

    & .checkmark:after {
      left: 5px;
      top: 2px;
      width: 5px;
      height: 10px;
      border: solid $white;
      border-width: 0 3px 3px 0;
      transform: rotate(45deg);
    }
  }
}

input#genres {
  cursor: pointer;
  height: 20px;
  width: 20px;

  &:focus {
    border: none;
  }
}

input#genre {
  text-transform: capitalize;
}

.selectorIcon:after {
  content: "â€º";
  position: absolute;
  right: -20px;
  top: 15px;
  bottom: 10px;
  font-size: 60px;
  color: $pink;
  transform: rotate(90deg);
}
