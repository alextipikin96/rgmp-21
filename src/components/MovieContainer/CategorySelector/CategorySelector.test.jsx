import {render, fireEvent, screen} from "@testing-library/react";
import CategorySelector from "./CategorySelector";
import React from "react";

describe("CategorySelector", () => {

    let mockHandler = jest.fn();

    it("should render properly",() => {
        const tree = render(<CategorySelector selectedCategories={[]} chooseCategoryHandler={mockHandler} />);

        expect(tree).toMatchSnapshot();
    });

    it("should call handler on change",() => {
        render(<CategorySelector selectedCategories={[]} chooseCategoryHandler={mockHandler} />);
        const checkbox = screen.getByDisplayValue("Horror");

        fireEvent.click(checkbox);

        expect(mockHandler).toBeCalledTimes(1);
    });

    it("should change visibility",() => {
        render(<CategorySelector selectedCategories={[]} chooseCategoryHandler={mockHandler} />);

        const categoryList = document.querySelector("ul");
        expect(categoryList.hidden).toBeTruthy();

        fireEvent.click(document.getElementsByClassName("inputWrapper")[0]);

        expect(categoryList.hidden).toBeFalsy();
    });
})
