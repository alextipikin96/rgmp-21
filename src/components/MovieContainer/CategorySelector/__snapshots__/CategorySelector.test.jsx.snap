// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategorySelector should render properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <label
        form="genre"
      >
        genre
      </label>
      <div
        class="inputWrapper"
      >
        <div
          class="selectorIcon"
        />
        <input
          disabled=""
          id="genre"
          placeholder="Select genre"
          value=""
        />
      </div>
      <ul
        class="categoryList"
        hidden=""
      >
        <li
          class="categoryItem"
        >
          <div
            class="categoryItem__wrapper"
          >
            <input
              id="genres"
              name="genres"
              type="checkbox"
              value="Documentary"
            />
            <span
              class=""
            >
              Documentary
            </span>
          </div>
        </li>
        <li
          class="categoryItem"
        >
          <div
            class="categoryItem__wrapper"
          >
            <input
              id="genres"
              name="genres"
              type="checkbox"
              value="Comedy"
            />
            <span
              class=""
            >
              Comedy
            </span>
          </div>
        </li>
        <li
          class="categoryItem"
        >
          <div
            class="categoryItem__wrapper"
          >
            <input
              id="genres"
              name="genres"
              type="checkbox"
              value="Horror"
            />
            <span
              class=""
            >
              Horror
            </span>
          </div>
        </li>
        <li
          class="categoryItem"
        >
          <div
            class="categoryItem__wrapper"
          >
            <input
              id="genres"
              name="genres"
              type="checkbox"
              value="Crime"
            />
            <span
              class=""
            >
              Crime
            </span>
          </div>
        </li>
        <li
          class="categoryItem"
        >
          <div
            class="categoryItem__wrapper"
          >
            <input
              id="genres"
              name="genres"
              type="checkbox"
              value="Adventure"
            />
            <span
              class=""
            >
              Adventure
            </span>
          </div>
        </li>
        <li
          class="categoryItem"
        >
          <div
            class="categoryItem__wrapper"
          >
            <input
              id="genres"
              name="genres"
              type="checkbox"
              value="Science Fiction"
            />
            <span
              class=""
            >
              Science Fiction
            </span>
          </div>
        </li>
        <li
          class="categoryItem"
        >
          <div
            class="categoryItem__wrapper"
          >
            <input
              id="genres"
              name="genres"
              type="checkbox"
              value="Action"
            />
            <span
              class=""
            >
              Action
            </span>
          </div>
        </li>
        <li
          class="categoryItem"
        >
          <div
            class="categoryItem__wrapper"
          >
            <input
              id="genres"
              name="genres"
              type="checkbox"
              value="Family"
            />
            <span
              class=""
            >
              Family
            </span>
          </div>
        </li>
        <li
          class="categoryItem"
        >
          <div
            class="categoryItem__wrapper"
          >
            <input
              id="genres"
              name="genres"
              type="checkbox"
              value="Animation"
            />
            <span
              class=""
            >
              Animation
            </span>
          </div>
        </li>
        <li
          class="categoryItem"
        >
          <div
            class="categoryItem__wrapper"
          >
            <input
              id="genres"
              name="genres"
              type="checkbox"
              value="Fantasy"
            />
            <span
              class=""
            >
              Fantasy
            </span>
          </div>
        </li>
        <li
          class="categoryItem"
        >
          <div
            class="categoryItem__wrapper"
          >
            <input
              id="genres"
              name="genres"
              type="checkbox"
              value="Drama"
            />
            <span
              class=""
            >
              Drama
            </span>
          </div>
        </li>
        <li
          class="categoryItem"
        >
          <div
            class="categoryItem__wrapper"
          >
            <input
              id="genres"
              name="genres"
              type="checkbox"
              value="Romance"
            />
            <span
              class=""
            >
              Romance
            </span>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <label
      form="genre"
    >
      genre
    </label>
    <div
      class="inputWrapper"
    >
      <div
        class="selectorIcon"
      />
      <input
        disabled=""
        id="genre"
        placeholder="Select genre"
        value=""
      />
    </div>
    <ul
      class="categoryList"
      hidden=""
    >
      <li
        class="categoryItem"
      >
        <div
          class="categoryItem__wrapper"
        >
          <input
            id="genres"
            name="genres"
            type="checkbox"
            value="Documentary"
          />
          <span
            class=""
          >
            Documentary
          </span>
        </div>
      </li>
      <li
        class="categoryItem"
      >
        <div
          class="categoryItem__wrapper"
        >
          <input
            id="genres"
            name="genres"
            type="checkbox"
            value="Comedy"
          />
          <span
            class=""
          >
            Comedy
          </span>
        </div>
      </li>
      <li
        class="categoryItem"
      >
        <div
          class="categoryItem__wrapper"
        >
          <input
            id="genres"
            name="genres"
            type="checkbox"
            value="Horror"
          />
          <span
            class=""
          >
            Horror
          </span>
        </div>
      </li>
      <li
        class="categoryItem"
      >
        <div
          class="categoryItem__wrapper"
        >
          <input
            id="genres"
            name="genres"
            type="checkbox"
            value="Crime"
          />
          <span
            class=""
          >
            Crime
          </span>
        </div>
      </li>
      <li
        class="categoryItem"
      >
        <div
          class="categoryItem__wrapper"
        >
          <input
            id="genres"
            name="genres"
            type="checkbox"
            value="Adventure"
          />
          <span
            class=""
          >
            Adventure
          </span>
        </div>
      </li>
      <li
        class="categoryItem"
      >
        <div
          class="categoryItem__wrapper"
        >
          <input
            id="genres"
            name="genres"
            type="checkbox"
            value="Science Fiction"
          />
          <span
            class=""
          >
            Science Fiction
          </span>
        </div>
      </li>
      <li
        class="categoryItem"
      >
        <div
          class="categoryItem__wrapper"
        >
          <input
            id="genres"
            name="genres"
            type="checkbox"
            value="Action"
          />
          <span
            class=""
          >
            Action
          </span>
        </div>
      </li>
      <li
        class="categoryItem"
      >
        <div
          class="categoryItem__wrapper"
        >
          <input
            id="genres"
            name="genres"
            type="checkbox"
            value="Family"
          />
          <span
            class=""
          >
            Family
          </span>
        </div>
      </li>
      <li
        class="categoryItem"
      >
        <div
          class="categoryItem__wrapper"
        >
          <input
            id="genres"
            name="genres"
            type="checkbox"
            value="Animation"
          />
          <span
            class=""
          >
            Animation
          </span>
        </div>
      </li>
      <li
        class="categoryItem"
      >
        <div
          class="categoryItem__wrapper"
        >
          <input
            id="genres"
            name="genres"
            type="checkbox"
            value="Fantasy"
          />
          <span
            class=""
          >
            Fantasy
          </span>
        </div>
      </li>
      <li
        class="categoryItem"
      >
        <div
          class="categoryItem__wrapper"
        >
          <input
            id="genres"
            name="genres"
            type="checkbox"
            value="Drama"
          />
          <span
            class=""
          >
            Drama
          </span>
        </div>
      </li>
      <li
        class="categoryItem"
      >
        <div
          class="categoryItem__wrapper"
        >
          <input
            id="genres"
            name="genres"
            type="checkbox"
            value="Romance"
          />
          <span
            class=""
          >
            Romance
          </span>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
